android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -511,7 +538,17 @@ public void setIncognito(WebView view, boolean enabled) {
   }
 
   @ReactProp(name = "source")
-  public void setSource(WebView view, @Nullable ReadableMap source) {
+  public void setSource(RNCWebView wrapper, @Nullable ReadableMap source) {
+    WebView view = wrapper.getWebView();
+
+    // TODO Note: If the 'source' prop is changed and the webView has a webViewKey
+    // the webview will not reload.
+    // We can fix this by checking to see by caching the previous source map
+    // and checking if the value has changed.
+    if (wrapper.getWebView().webViewKey != null && view.getUrl() != null) {
```

Comment:
See comment above, we can fix this is an a follow-up PR. We can do a shallow comparison of all the key/value pairs inside the `source` hash-map of the `new` source and the `old` source.

Reply:
Sounds good. Can you add a subtask in https://app.asana.com/0/1201254368985004/1202185487115885 for this?